odoo/addons/base/models/res_partner.py

```diff
@@ -237,6 +237,12 @@ def default_get(self, default_fields):
         ('check_name', "CHECK( (type='contact' AND name IS NOT NULL) or (type!='contact') )", 'Contacts require a name'),
     ]
 
+    @api.onchange('company_id')
+    def _compute_allowed_salesperson(self):
+        for partner in self:
+            if(partner.company_id not in partner.user_id.company_ids):
+                partner.user_id = False
+
```

Comment:
You've forgotten to remove this unused code ;)

Reply:
I left it there so that when you modify the company and the current user is not allowed in the new company it removes it.

Reply:
@kitan191